### PR TITLE
Clean up network setting docs

### PR DESCRIPTION
**Pull request by DaveCTurner**
- Add types for all params
- Remove mention of JDKs before 11
- Clarify some wording

Co-authored-by: Stef Nestor <steffanie.nestor@gmail.com>

### CODE REVIEW

File: docs/reference/modules/http.asciidoc [hunk 1]
```diff
@@ -95,109 +95,117 @@ IMPORTANT: A wildcard (`*`) is a valid value but is considered a security risk,
...
 `http.tcp.keep_count`::
-(<<static-cluster-setting,Static>>) Configures the `TCP_KEEPCNT` option for this socket, which
-determines the number of unacknowledged TCP keepalive probes that may be
-sent on a connection before it is dropped. Defaults to `network.tcp.keep_count`,
-which uses the system default. Only applicable on Linux and macOS, and
-requires Java 11 or newer.
+(<<static-cluster-setting,Static>>, integer)
+Configures the `TCP_KEEPCNT` option for HTTP sockets, which determines the
+number of unacknowledged TCP keepalive probes that may be sent on a connection
+before it is dropped. Defaults to `network.tcp.keep_count`, which uses the
```
Comment (lockewritesdocs) [inline docs/reference/modules/http.asciidoc:174]:
```suggestion
before it is dropped. Defaults to `-1`, which uses the
```

Comment (lockewritesdocs) [reply docs/reference/modules/http.asciidoc:174]: This is an `integer` setting, which I believe defaults to `-1`.

File: docs/reference/modules/http.asciidoc [hunk 2]
```diff
@@ -95,109 +95,117 @@ IMPORTANT: A wildcard (`*`) is a valid value but is considered a security risk,
...
+the address can be reused or not. Defaults to `false` on Windows and `true`
+otherwise.
 
 `http.tcp.send_buffer_size`::
-(<<static-cluster-setting,Static>>)
-The size of the TCP send buffer (specified with <<size-units,size units>>).
-Defaults to `network.tcp.send_buffer_size`.
+(<<static-cluster-setting,Static>>, <<byte-units,byte value>>)
+The size of the TCP send buffer for HTTP traffic. Defaults to
+`network.tcp.send_buffer_size`.
```
Comment (lockewritesdocs) [inline docs/reference/modules/http.asciidoc:191]: This setting accepts a `byte value`, but the default is listed as a `string`.

File: docs/reference/modules/http.asciidoc [hunk 3]
```diff
@@ -95,109 +95,117 @@ IMPORTANT: A wildcard (`*`) is a valid value but is considered a security risk,
...
 
 `http.tcp.receive_buffer_size`::
-(<<static-cluster-setting,Static>>)
-The size of the TCP receive buffer (specified with <<size-units,size units>>).
-Defaults to `network.tcp.receive_buffer_size`.
+(<<static-cluster-setting,Static>>, <<byte-units,byte value>>)
+The size of the TCP receive buffer for HTTP traffic. Defaults to
+`network.tcp.receive_buffer_size`.
```
Comment (lockewritesdocs) [inline docs/reference/modules/http.asciidoc:196]: This setting accepts a `byte value`, but the default is listed as a `string`.

File: docs/reference/modules/transport.asciidoc [hunk 4]
```diff
@@ -36,92 +36,91 @@ different addresses for publishing and binding, and you also require different
...
-determines the time in seconds that a connection must be idle before
-starting to send TCP keepalive probes. Defaults to `network.tcp.keep_idle` if set,
-or the system default otherwise.
-This value cannot exceed `300` seconds. In cases where the system default
-is higher than `300`, the value is automatically lowered to `300`. Only applicable on
-Linux and macOS, and requires Java 11 or newer.
+(<<static-cluster-setting,Static>>, integer)
+Configures the `TCP_KEEPIDLE` option for transport sockets, which determines
+the time in seconds that a connection must be idle before starting to send TCP
+keepalive probes. Defaults to `network.tcp.keep_idle` if set, or the system
```
Comment (lockewritesdocs) [inline docs/reference/modules/transport.asciidoc:82]:
```suggestion
keepalive probes. Defaults to `-1` if set, or the system
```

Comment (lockewritesdocs) [reply docs/reference/modules/transport.asciidoc:82]: I think that this `integer` setting defaults to `-1`.

File: docs/reference/modules/transport.asciidoc [hunk 5]
```diff
@@ -36,92 +36,91 @@ different addresses for publishing and binding, and you also require different
...
-Configures the `TCP_KEEPINTVL` option for this socket,
-which determines the time in seconds between sending TCP keepalive probes.
-Defaults to `network.tcp.keep_interval` if set, or the system default otherwise.
-This value cannot exceed `300` seconds. In cases where the system default is higher than `300`,
-the value is automatically lowered to `300`. Only applicable on Linux and macOS,
-and requires Java 11 or newer.
+(<<static-cluster-setting,Static>>, integer)
+Configures the `TCP_KEEPINTVL` option for transport sockets, which determines
+the time in seconds between sending TCP keepalive probes. Defaults to
+`network.tcp.keep_interval` if set, or the system default otherwise. This value
```
Comment (lockewritesdocs) [inline docs/reference/modules/transport.asciidoc:91]:
```suggestion
`-1` if set, or the system default otherwise. This value
```

Comment (lockewritesdocs) [reply docs/reference/modules/transport.asciidoc:91]: I think that this `integer` setting defaults to `-1`.

File: docs/reference/modules/transport.asciidoc [hunk 6]
```diff
@@ -36,92 +36,91 @@ different addresses for publishing and binding, and you also require different
...
-(<<static-cluster-setting,Static>>)
-Configures the `TCP_KEEPCNT` option for this socket, which
-determines the number of unacknowledged TCP keepalive probes that may be
-sent on a connection before it is dropped. Defaults to `network.tcp.keep_count`
-if set, or the system default otherwise. Only applicable on Linux and macOS, and
-requires Java 11 or newer.
+(<<static-cluster-setting,Static>>, integer)
+Configures the `TCP_KEEPCNT` option for transport sockets, which determines the
+number of unacknowledged TCP keepalive probes that may be sent on a connection
+before it is dropped. Defaults to `network.tcp.keep_count` if set, or the
```
Comment (lockewritesdocs) [inline docs/reference/modules/transport.asciidoc:100]:
```suggestion
before it is dropped. Defaults to `-1` if set, or the
```

Comment (lockewritesdocs) [reply docs/reference/modules/transport.asciidoc:100]: I think that this `integer` setting defaults to `-1`.

File: docs/reference/modules/transport.asciidoc [hunk 7]
```diff
@@ -36,92 +36,91 @@ different addresses for publishing and binding, and you also require different
...
+whether {wikipedia}/Nagle%27s_algorithm[TCP no delay] is enabled. Defaults to
+`true`.
 
 `transport.tcp.reuse_address`::
-(<<static-cluster-setting,Static>>)
-Should an address be reused or not. Defaults to `network.tcp.reuse_address`.
+(<<static-cluster-setting,Static>>, boolean)
+Configures the `SO_REUSEADDR` option for network sockets, which determines
+whether the address can be reused or not. Defaults to
+`network.tcp.reuse_address`.
```
Comment (lockewritesdocs) [inline docs/reference/modules/transport.asciidoc:113]:
```suggestion
`true`.
```

Comment (lockewritesdocs) [reply docs/reference/modules/transport.asciidoc:113]: I think that this `boolean` setting defaults to `true`.

File: docs/reference/modules/transport.asciidoc [hunk 8]
```diff
@@ -36,92 +36,91 @@ different addresses for publishing and binding, and you also require different
...
 
 `transport.tcp.send_buffer_size`::
-(<<static-cluster-setting,Static>>)
-The size of the TCP send buffer (specified with <<size-units,size units>>).
-Defaults to `network.tcp.send_buffer_size`.
+(<<static-cluster-setting,Static>>, <<byte-units,byte value>>)
+The size of the TCP send buffer for transport traffic. Defaults to
+`network.tcp.send_buffer_size`.
```
Comment (lockewritesdocs) [inline docs/reference/modules/transport.asciidoc:118]:
```suggestion
`-1`, which means to use the system default.
```

Comment (lockewritesdocs) [reply docs/reference/modules/transport.asciidoc:118]: I think that this `byte value` setting defaults to `-1`.

File: docs/reference/modules/transport.asciidoc [hunk 9]
```diff
@@ -36,92 +36,91 @@ different addresses for publishing and binding, and you also require different
...
 
 `transport.tcp.receive_buffer_size`::
-(<<static-cluster-setting,Static>>)
-The size of the TCP receive buffer (specified with <<size-units,size units>>).
-Defaults to `network.tcp.receive_buffer_size`.
+(<<static-cluster-setting,Static>>, <<byte-units,byte value>>)
+The size of the TCP receive buffer for transport traffic. Defaults to
+`network.tcp.receive_buffer_size`.
```
Comment (lockewritesdocs) [inline docs/reference/modules/transport.asciidoc:123]:
```suggestion
`-1`, which means to use the system default.
```

Comment (lockewritesdocs) [reply docs/reference/modules/transport.asciidoc:123]: I think that this `byte value` setting defaults to `-1`.

**Review by lockewritesdocs — COMMENTED** [review]
Left some suggested changes based on the setting type and the default values.

**Comment by DaveCTurner** [comment]
Thanks @lockewritesdocs. The `transport.*` and `http.*` settings largely fall back to the `network.*` ones so the changes you've suggested are not quite accurate. For instance if you set `network.tcp.keep_count: 5` then `transport.tcp.keep_count` defaults to 5, not to -1. That's why we say that they default to the fallback setting in each case.

File: docs/reference/modules/http.asciidoc [hunk 10]
```diff
@@ -95,109 +95,117 @@ IMPORTANT: A wildcard (`*`) is a valid value but is considered a security risk,
...
 `http.detailed_errors.enabled`::
-(<<static-cluster-setting,Static>>)
-If `true`, enables the output of detailed error messages and stack traces in the response output. Defaults to `true`.
-+
-If `false`, use the `error_trace` parameter to <<common-options-error-options,enable stack traces>> and return detailed error messages. Otherwise, only a simple message will be returned.
+(<<static-cluster-setting,Static>>, boolean)
+Configures whether detailed errors may be returned in HTTP responses. Defaults
+to `true`, which means that HTTP requests which include
+<<common-options-error-options,`?error_trace` URI parameter>> will return a
+detailed error message including a stack trace if they encounter an exception.
```
[the remark this answers is not in the document]
Comment (DaveCTurner) [reply docs/reference/modules/http.asciidoc:133]: I think `may be` is correct here. By default we don't return detailed errors even if `http.detailed_errors.enabled: true` - this setting just permits the use of the `?error_trace` parameter.

**Review by lockewritesdocs — APPROVED** [review]
Thanks for the explanation @DaveCTurner. I resolved the comments relating to suggested changes on those settings. Merge away 🦖

**Comment by lockewritesdocs** [comment]
@DaveCTurner, it looks like the 8.3 backport failed.

**Comment by DaveCTurner** [comment]
Thanks, yes, I did a custom bespoke artisanal backport to 8.3 in a42fc90fbbf3ced5c143f349bd42cbc42b367161.

**Comment by lockewritesdocs** [comment]
> Thanks, yes, I did a custom bespoke artisanal backport to 8.3 in [a42fc90](https://github.com/elastic/elasticsearch/commit/a42fc90fbbf3ced5c143f349bd42cbc42b367161).

I'm assuming that you physically lifted the changes into the 8.3 branch, and then used a vintage bellows to propagate them through the code base.

**Pull request merged**